system/core/tags.php

```diff
@@ -3,6 +3,64 @@
 	// - implement error notification if new tag exists
 	// - implement tag management page: edit, translate, remove or merge tags
 	// - implement tagcontainer pagetype
+	class HyphaTags {
+		static function findTagByLabel($lang, $label) {
+			/** @var HyphaDomElement $hyphaXml */
+			global $hyphaXml;
+			$path = 'hypha/tagList/tag/language[@id=' . xpath_encode($lang) . ' and @label=' . xpath_encode($label) . ']';
+			$node = $hyphaXml->findXPath($path)->parent()->first();
+			if (!$node)
```

Comment:
Would suggest a ternary (as a non blocking review suggestion :-) )

Reply:
Here I think it would indeed help readability, so I've made this a ternary if.